packages/core/src/errors.ts

```diff
@@ -12,9 +12,35 @@ export enum ProviderFailure {
 
 const formatMessage = (reason: string, detail?: string) => reason + (detail ? ` (${detail})` : '');
 
-export class ProviderError<InnerError = unknown> extends CustomError {
-  constructor(public reason: ProviderFailure, public innerError?: InnerError, public detail?: string) {
-    super(formatMessage(reason, detail));
+export class ErrorWithInnerError<InnerError = unknown> extends CustomError {
```

Comment:
Just an idea, original name is ok too
```suggestion
export class ComposableError<InnerError = unknown> extends CustomError {
```

Reply:
Hi @mkazlauskas , 18a531dea and 18a531dea should fix this